trustmanager/keyfilestore.go

```diff
@@ -87,9 +207,17 @@ func (s *KeyFileStore) ExportKey(name string) ([]byte, error) {
 }
 
 // ImportKey imports the private key in the encrypted bytes into the keystore
-// with the given key ID and alias.
-func (s *KeyFileStore) ImportKey(pemBytes []byte, alias string) error {
-	return importKey(s, s.Retriever, s.cachedKeys, alias, pemBytes)
+// with the given key ID, role, and gun.
+func (s *KeyFileStore) ImportKey(pemBytes []byte, role, gun string) error {
+	if data.IsDelegation(role) {
```

Comment:
`KeyFilestore.AddKey` sets the gun to empty `keyInfo.Role == data.CanonicalRootRole || data.IsDelegation(keyInfo.Role) || !data.ValidRole(keyInfo.Role)` - would the same check make sense here?